core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts

```diff
@@ -29,26 +29,63 @@ const acceptedBrowserAlerts = [
   'This action is irreversible. Are you sure?'
 ];
 
-module.exports = class baseUser {
-  constructor() {
-    this.page;
-    this.browserObject;
-    this.userHasAcceptedCookies = false;
-  }
+interface IUserProps {
+  page: Page;
+  browserObject: Browser;
+  userHasAcceptedCookies: boolean;
+  username: string;
+  email: string;
+}
+
+export interface IBaseUser extends IUserProps {
+  openBrowser: () => Promise<Page>;
+  signInWithEmail: (email: string) => Promise<void>;
+  signUpNewUser: (userName: string, signInEmail: string) => Promise<void>;
+  reloadPage: () => Promise<void>;
+  clickOn: (selector: string) => Promise<void>;
+  type: (selector: string, text: string) => Promise<void>;
+  select: (selector: string, option: string) => Promise<void>;
+  goto: (url: string) => Promise<void>;
+  uploadFile: (filePath: string) => Promise<void>;
+  logout: () => Promise<void>;
+  closeBrowser: () => Promise<void>;
+}
+
+export class BaseUser implements IBaseUser {
+  page!: Page;
+  browserObject!: Browser;
+  userHasAcceptedCookies: boolean = false;
+  email: string = '';
+  username: string = '';
+
+  constructor() {}
 
   /**
    * This is a function that opens a new browser instance for the user.
-   * @returns {Promise<puppeteer.Page>} - Returns a promise that resolves
-   * to a Page object controlled by Puppeteer.
    */
-  async openBrowser() {
+  async openBrowser(): Promise<Page> {
+    const args: string[] = [
+      '--start-fullscreen',
+      '--use-fake-ui-for-media-stream'
+    ];
+
+    const headless = (
+      process.env.HEADLESS ?
+      process.env.HEADLESS === 'true' :
+      testConstants.DEFAULT_IS_HEADLESS);
+    /**
+     * Here we are disabling the site isolation trials because it is causing
+     * tests to fail while running in non headless mode (see
+     * https://github.com/puppeteer/puppeteer/issues/7050).
+     */
+    if (!headless) {
+      args.push('--disable-site-isolation-trials');
+    }
+
     await puppeteer
       .launch({
-        /** TODO(#17761): Right now some acceptance tests are failing on
```

Comment:
Should we close this issue if it's no longer valid (or did you just not copy the TODO over)?